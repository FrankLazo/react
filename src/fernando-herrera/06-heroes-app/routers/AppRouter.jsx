import { useContext } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import AuthContext from '../auth/AuthContext';
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoutes from './DashboardRoutes';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = () =>
{
	const { user: { logged } } = useContext(AuthContext);

	return (
		<Router>
			<div className="min-height-full has-background-light">
				<Switch>
					<PublicRoute exact path="/login" component={ LoginScreen }  isAuthenticated={ logged } />
					<PrivateRoute path="/" component={ DashboardRoutes } isAuthenticated={ logged } />
				</Switch>
			</div>
		</Router>
	);
};

export default AppRouter;
