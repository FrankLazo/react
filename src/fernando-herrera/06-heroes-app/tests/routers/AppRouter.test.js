import { mount } from 'enzyme';
import AuthContext from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('<AppRouter /> tests', () => {
	const contextValue = {
		user: {
			logged: false,
		},
		dispatch: jest.fn(),
	};

	test('should show login if it is not authenticated', () => {
		const wrapper = mount(
			<AuthContext.Provider value={{ ...contextValue }} >
				<AppRouter />
			</AuthContext.Provider>
		);

		// console.log(wrapper.html());
		expect(wrapper).toMatchSnapshot();
	});

	test('should show marvel component if it is authenticated', () => {
		const contextValue = {
			user: {
				name: 'Frank',
				logged: true,
			},
			dispatch: jest.fn(),
		};

		const wrapper = mount(
			<AuthContext.Provider value={{ ...contextValue }} >
				<AppRouter />
			</AuthContext.Provider>
		);

		// console.log(wrapper.html());
		expect(wrapper.find('.navbar').exists()).toBe(true);
	});
});
