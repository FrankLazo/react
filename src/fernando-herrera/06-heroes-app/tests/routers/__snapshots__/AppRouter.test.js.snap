// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> tests should show login if it is not authenticated 1`] = `
<div
  className="min-height-full has-background-light"
>
  <div>
    <div
      className="min-height-full content has-background-light p-4"
    >
      <h1
        className="has-text-primary"
      >
        Login
      </h1>
      <hr
        className="has-background-grey-lighter"
      />
      <button
        className="button is-primary"
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
